backend/app/cmd/server.go

```diff
@@ -859,7 +859,10 @@ func (s *ServerCommand) addAuthProviders(authenticator *auth.Service) error {
 				return false, nil
 			}
 			return true, nil
-		}))
+		}),
+			func(user string, r *http.Request) string {
```

Comment:
I propose to add a comment here, something like "Custom user ID generator, used to distinguish anonymous users coming from different IPs".